src/firmware/host/types/snp.rs

```diff
@@ -122,6 +126,19 @@ impl CertTableEntry {
     pub fn new(cert_type: CertType, data: Vec<u8>) -> Self {
         Self { cert_type, data }
     }
+
+    /// Builds a Kernel formatted CertTable for sending the certificate content to the PSP.
+    pub fn cert_table_to_vec_bytes(table: &[Self]) -> Result<Vec<u8>, CertError> {
+        FFI::types::CertTableEntry::uapi_to_vec_bytes(&table.to_vec())
+    }
+
+    /// Takes in bytes in kernel CertTable format and returns in user API CertTable format.
+    pub fn vec_bytes_to_cert_table(mut bytes: Vec<u8>) -> Result<Vec<Self>, CertError> {
```

Comment:
We should probably just pass this by reference, so we aren't doing a move/copy here.

```suggestion
    pub fn vec_bytes_to_cert_table(bytes: &mut [u8]) -> Result<Vec<Self>, CertError> {
```